subsys/net/lib/http/http_client.c

```diff
@@ -120,9 +121,16 @@ static int http_send_data(int sock, char *send_buf,
 
 static int http_flush_data(int sock, const char *send_buf, size_t send_buf_len)
```

Comment:
I'd suggest making this "size_t *send_buf_pos" so that it's consistent with http_send_data and you don't need to remember to zero it at each call site.